phantom_token.c

```diff
@@ -488,129 +495,17 @@ static ngx_int_t handler(ngx_http_request_t *request)
-    }
-
-    if (error_code_provided)
-    {
-        error_code_len = ngx_strlen(error_code);
-        bearer_data_size += ERROR_CODE_PREFIX_SIZE + error_code_len + TOKEN_SUFFIX_SIZE;
-    }
-
-    if ((realm_provided && scopes_provided) || (realm_provided && error_code_provided) || (scopes_provided && error_code_provided))
-    {
-        bearer_data_size += TOKEN_SEPARATER_SIZE;
-        append_one_comma = true;
-
-        if (realm_provided && scopes_provided && error_code_provided)
-        {
-            bearer_data_size += TOKEN_SEPARATER_SIZE;
-            append_two_commas = true;
-        }
-    }
-
-    u_char *bearer_data = ngx_pnalloc(request->pool, bearer_data_size);
-
-    if (bearer_data == NULL)
-    {
-        return NGX_HTTP_INTERNAL_SERVER_ERROR;
-    }
-
-    u_char *p = ngx_cpymem(bearer_data, BEARER, BEARER_SIZE);
-
-    if (realm_provided)
-    {
-        p = ngx_cpymem(p, REALM_PREFIX, REALM_PREFIX_SIZE);
-        p = ngx_cpymem(p, module_location_config->realm.data, module_location_config->realm.len);
-        p = ngx_cpymem(p, TOKEN_SUFFIX, TOKEN_SUFFIX_SIZE);
-
-        if (append_one_comma)
-        {
-            p = ngx_cpymem(p, TOKEN_SEPARATER, TOKEN_SEPARATER_SIZE);
-        }
-    }
-
-    if (scopes_provided)
-    {
-        p = ngx_cpymem(p, SCOPE_PREFIX, SCOPE_PREFIX_SIZE);
-        p = ngx_cpymem(p, module_location_config->space_separated_scopes.data, module_location_config->space_separated_scopes.len);
-        p = ngx_cpymem(p, TOKEN_SUFFIX, TOKEN_SUFFIX_SIZE);
-
-        if (append_one_comma || append_two_commas)
-        {
-            p = ngx_cpymem(p, TOKEN_SEPARATER, TOKEN_SEPARATER_SIZE);
-        }
-    }
-
-    if (error_code_provided)
-    {
-        p = ngx_cpymem(p, ERROR_CODE_PREFIX, ERROR_CODE_PREFIX_SIZE);
-        p = ngx_cpymem(p, error_code, error_code_len);
-        p = ngx_cpymem(p, TOKEN_SUFFIX, TOKEN_SUFFIX_SIZE);
-    }
-
-    if (!scopes_provided && !realm_provided && !error_code_provided)
-    {
-        // Only 'Bearer' is being sent. Replace the space at the end of BEARER with a null byte.
-        *(p - 1) = '\0';
-    }
-    else
-    {
-        *p = '\0';
-    }
-
-    request->headers_out.www_authenticate->hash = 1;
-    ngx_str_set(&request->headers_out.www_authenticate->key, "WWW-Authenticate");
-    request->headers_out.www_authenticate->value.data = bearer_data;
-    request->headers_out.www_authenticate->value.len = ngx_strlen(bearer_data);
-
-    assert(request->headers_out.www_authenticate->value.len <= bearer_data_size);
-
-    return write_error_response(request, NGX_HTTP_UNAUTHORIZED, module_location_config);
-}
-
 static ngx_int_t introspection_response_handler(ngx_http_request_t *request, void *data,
                                                 ngx_int_t introspection_subrequest_status_code)
 {
     phantom_token_module_context_t *module_context = (phantom_token_module_context_t*)data;
```

Comment:
This is exactly the same logic branches as previously but I moved variables to the top and added some comments.